lib/manager/flux/extract.ts

```diff
@@ -0,0 +1,123 @@
+import { loadAll } from 'js-yaml';
+import { HelmDatasource } from '../../datasource/helm';
+import { logger } from '../../logger';
+import { SkipReason } from '../../types';
+import { readLocalFile } from '../../util/fs';
+import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
+import type {
+  FluxManifest,
+  FluxResource,
+  HelmRelease,
+  HelmRepository,
+} from './types';
+
+function readManifest(content: string): FluxManifest {
+  const manifest: FluxManifest = { releases: [], repositories: [] };
+  let resources: FluxResource[];
+  try {
+    resources = loadAll(content, null, { json: true }) as FluxResource[];
+  } catch (err) {
+    logger.debug({ err }, 'Failed to parse Flux manifest');
+    return null;
+  }
+
+  // It's possible there are other non-Flux HelmRelease/HelmRepository CRs out there, so we filter based on apiVersion.
+  for (const resource of resources) {
+    switch (resource?.kind) {
+      case 'HelmRelease':
+        if (
+          resource.apiVersion?.startsWith('helm.toolkit.fluxcd.io/') &&
```

Comment:
Move the API version check before the switch to make code less verbose

Reply:
Can you clarify? The arguments to `startsWith` are different since the resources are in different API groups so I'm not sure how to make this code less verbose. 🤔

Reply:
@viceice were you thinking for DRY reasons? Because the two strings are difference, so I'm also confused